### PR TITLE
ShA/infinite pagination

### PR DESCRIPTION
**Pull request by AlekseiiShevchuk**
add infinite pagination for Topics, My-Topics and Votes

### CODE REVIEW

File: resources/assets/js/app/views/topics/topicCollection.js [hunk 1]
```diff
@@ -3,5 +3,58 @@ var Marionette = require('backbone.marionette');
 var topicItem = require('./topicItem');
 
 module.exports = Marionette.CollectionView.extend({
-  childView: topicItem
+    childView: topicItem,
+
+    initialize: function () {
+
+            $(window).on('scroll', this.fetchPage.bind(this));
```
Comment (TatsuUkraine) [inline resources/assets/js/app/views/topics/topicCollection.js]: This bind will fetch data even if you don't need this

**Comment by AlekseiiShevchuk** [comment]
i`ve done code refactor

File: app/Http/Controllers/TopicController.php [hunk 2]
```diff
@@ -20,15 +20,15 @@ class TopicController extends ApiController
     protected $tagIds = [];
 
     /**
-    /**
+     * /**
```
Comment (TatsuUkraine) [inline app/Http/Controllers/TopicController.php]: remove this row

**Comment by TatsuUkraine** [comment]
use getCollection() instead of items() in pagination Class

File: resources/assets/js/app/instances/paginateableCollectionView.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+var Marionette = require('backbone.marionette');
+
+module.exports = Marionette.CollectionView.extend({
+
+    initialize: function () {
+
+        $(window).on('scroll', this.fetchPage.bind(this));
```
Comment (TatsuUkraine) [inline resources/assets/js/app/instances/paginateableCollectionView.js]: this event will be binded again and again on each view creation, you need or unbind it on view destroy or create event mediator with Radio and use Radio to listen scroll events

**Comment by AlekseiiShevchuk** [comment]
done

File: resources/assets/js/app/instances/paginateableCollectionView.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    },
+    onDestroy: function(){
+        $(window).off('scroll', this.fetchPage.bind(this));
```
Comment (TatsuUkraine) [inline resources/assets/js/app/instances/paginateableCollectionView.js]: remove callback, and leave just $(window).off('scroll');
with callback this event still present

**Comment by AlekseiiShevchuk** [comment]
Done

**Comment by AlekseiiShevchuk** [comment]
Done

**Comment by TatsuUkraine** [comment]
when empty collection is fetched it removes meta, add logic in parse to merge meta data from server with meta in collection if options.remove = false

**Comment by AlekseiiShevchuk** [comment]
Done, tested in chrome, firefox and IE

**Pull request merged**